### DISCUSSION TITLE
Jedis version-2.7.2 throw ClassCastException: [B cannot be cast to XXX

### DISCUSSION THREAD

**Discussion by flyletc** [xetorthio/jedis#2653] [Q&A]
The connection encapsulates the network output stream, in which a buffer is built. Therefore, when an exception occurs, there are still commands that were not sent or incomplete sent last time in the buffer. At this time, no processing is done. The connection is directly returned to the connection pool. Then, when the connection is reused to execute the next command, The command not sent last time will be sent together, so the above error "wrong return value type" will appear;

**Comment by sazzad16** [comment]
@flyletc 2.7.2 was released almost 6.5 years ago. Please use latest version.

**Reply by flyletc** [reply]
Thank you,but I want to confirm whether this problem has been solved in the new version，because I saw someone encounter similar problems in the new version

**Reply by sazzad16** [reply]
In most cases, similar issues turned out to be due to wrongful use.

Still, there have been several commits and releases. You should be the one to check/test whether it is enough for you/your application.

The approach should be using a recent version and if there is an issue report with reproducible code and at least stacktrace.

**Reply by flyletc** [reply]
Thank you  very much,the problem has been solved
